### PR TITLE
fix: no warning for missing dependencies if no component exists

### LINKED ISSUE #617: Dependency warning for root component only
If I create an SBOM with only a root component as follows:

```python
from cyclonedx.model.bom import Bom, BomMetaData
from cyclonedx.model.component import Component
from cyclonedx.output.json import JsonV1Dot6

bom = Bom(metadata=BomMetaData(component=Component(name="test",version="1.2")))
print(JsonV1Dot6(bom).output_as_string())
```

I get the warning:
```bash
/miniforge3/envs/cyclonedx/lib/python3.11/site-packages/cyclonedx/model/bom.py:667: UserWarning: The Component this BOM is describing None has no defined dependencies which means the Dependency Graph is incomplete - you should add direct dependencies to this "root" Component to complete the Dependency Graph data.
  warn(
```

I can validate this SBOM with sbom-utility and to my understanding this is a valid SBOM. There should be no warning, or?

I use version cyclonedx-python-lib version 7.3.4 and Python 3.11.

### PR DESCRIPTION
**Pull request by weichslgartner**
Adds a condition to check if the root component is the only component and then doesn't issue the warning. Closes #617

### CODE REVIEW

File: tests/test_model_bom.py [hunk 1]
```diff
@@ -139,6 +139,15 @@ def test_empty_bom(self) -> None:
         self.assertFalse(bom.services)
         self.assertFalse(bom.external_references)
 
+    def test_root_component_only_bom(self) -> None:
+        with warnings.catch_warnings():
```
Comment (jkowalleck) [inline tests/test_model_bom.py:143]: please use python `unittest` framework mechanisms: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns

**Review by jkowalleck — CHANGES_REQUESTED** [review]

File: tests/test_model_bom.py (hunk 1 above)
Comment (weichslgartner) [reply tests/test_model_bom.py:143]: I only found assertWarns, but I need assertNotWarns which seems not to exist: https://discuss.python.org/t/add-assertnotwarns-method-to-unittest-testcase/27351
The action parameter in catch_warning was only added in Python 3.11: https://docs.python.org/3/library/warnings.html#warnings.catch_warnings
How should I proceed here?

**Comment by jkowalleck** [comment]
@madpah , what do you think about this change?

File: tests/test_model_bom.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (weichslgartner) [reply tests/test_model_bom.py]: done

**Review by jkowalleck — APPROVED** [review]

**Pull request merged**